lib/controller.dart

```diff
@@ -367,27 +389,40 @@ void command(UIAction action, Data data) {
       break;
     case UIAction.toggleActiveFilter:
       var d = data as ToggleActiveFilterData;
+      var filterDropdownID = generateFilterDropdownID(d.key);
+      var comparisonDropdownID = generateComparisonFilterDropdownID(d.key);
```

Comment:
I don't think the controller should be doing this kind of translation for the view (ie. computing the ID of the view elements). If needed, the view can keep track of the mapping key -> DOM element, or you can just pass the key to the view, which can then compute the appropriate ID?

Reply:
That makes sense - have changed it to make the controller just pass the filter key and the view computes to toggle on/off the drop-downs. Also removed the previously used utils for generating the IDs.